src/cache/RioCache.js

```diff
@@ -15,19 +15,22 @@ function isCollection(entity) {
 
 /**
  * Compare cached entity modification time with current.
- * If any argument is not provided, cache will be considered invalid.
+ * It is not expected to require older version of references then current in the state,
+ * thus if modification time is not strictly equal it is considered different.
+ * This is preliminary change to get cache work as it will with reference comparator.
  *
  * @param cachedModificationTime
  * @param currentModificationTime
  * @returns {boolean}
  */
 function isCacheValid(cachedModificationTime, currentModificationTime) {
-  return cachedModificationTime >= currentModificationTime;
```

Comment:
In the first iteration we were focused on `modifiedTimestamp` as relevant thing to determine if reference is changed and for cache it had sense to return always newer version, not caching older. Now when we are looking different at cache, taking reference as relevant thing, concept of newer and older reference is lost, it is only important if this is the same reference as cached.